mcstatus/server.py

```diff
@@ -59,6 +59,18 @@ class JavaServer(MCServer):
 
     DEFAULT_PORT = 25565
 
+    def __init__(self, host: str, port: int | None = None, timeout: float = 3, query_port: int | None = None):
+        """
+        :param host: The host/ip of the minecraft server.
+        :param port: The port that the server is on.
+        :param timeout: The timeout in seconds before failing to connect.
+        :param query_port: Typically the same as ``port`` but can be different.
+        """
+        super().__init__(host, port, timeout)
+        if query_port is None:
+            query_port = port or self.DEFAULT_PORT
+        self.query_port = query_port
```

Comment:
The only thing I miss here is the initialization checks that come with `Address` for the regular port, e.g. the valid port range.

With this, init will pass even on port numbers that are outside of the correct range, only failing when query is called, where the address object is created, which is inconsistent with the behavior of `port` being wrong.

I'm not sure how much we want to care about this, the way I see it, there are 2 solutions for that if we do want to address it. One would be to just construct the address object and throw it away:
```suggestion
        if query_port is None:
            query_port = port or self.DEFAULT_PORT
        else:
            _ = Address(host, query_port)  # Make sure query_port is valid
        self.query_port = query_port
```

The other option would be to construct address and store it, instead of just storing the port. As discussed on the issue, that would bring in some redundancy though.

Reply:
I'll add the Address construct. Seems more consistent to do the check during init if it happens like this in other classes init too.